system/ubloxd/pigeond.py

```diff
@@ -41,16 +44,22 @@ def add_ubx_checksum(msg: bytes) -> bytes:
     B = (B + A) % 256
   return msg + bytes([A, B])
 
-def get_assistnow_messages(token: bytes) -> list[bytes]:
-  # make request
-  # TODO: implement adding the last known location
-  r = requests.get("https://online-live2.services.u-blox.com/GetOnlineData.ashx", params=urllib.parse.urlencode({
-    'token': token,
-    'gnss': 'gps,glo',
-    'datatype': 'eph,alm,aux',
-  }, safe=':,'), timeout=5)
-  assert r.status_code == 200, "Got invalid status code"
-  dat = r.content
+def get_assistnow_messages(token: bytes, assistance_path: str) -> list[bytes]:
+  if token:
+    # make request
+    # TODO: implement adding the last known location
+    r = requests.get("https://online-live2.services.u-blox.com/GetOnlineData.ashx", params=urllib.parse.urlencode({
+      'token': token,
+      'gnss': 'gps,glo',
+      'datatype': 'eph,alm,aux',
+    }, safe=':,'), timeout=5)
+    assert r.status_code == 200, "Got invalid status code"
+    dat = r.content
+  elif assistance_path:
+    with open(assistance_path, 'rb') as f:
```

Comment:
FYI, looks like we want to use the "offline" flavor for this https://gist.github.com/veproza/55ec6eaa612781ac29e7#u-blox-a-gps-offerings